ui/app/AppLayouts/Communities/views/CommunitySettingsView.qml

```diff
@@ -504,6 +519,62 @@ StatusSectionLayout {
         }
     }
 
+    StatusQUtils.ModelChangeTracker {
+        id: tokensModelChangesTracker
+
+        // Owner and TMaster token deployment states
+        property bool isOwnerTokenDeployed: false
+        property bool isTMasterTokenDeployed: false
+
+        // It will monitorize if Owner and/or TMaster token items are included in the `model` despite the deployment state
+        property bool ownerOrTMasterTokenItemsExist: false
+
+        function checkIfPrivilegedTokenItemsExist() {
+           return SQUtils.ModelUtils.contains(model, "name", PermissionsHelpers.ownerTokenNameTag + root.communityName) ||
+                  SQUtils.ModelUtils.contains(model, "name", PermissionsHelpers.tMasterTokenNameTag + root.communityName)
+        }
+
+        function reviewTokenDeployState(tagType, isOwner) {
+            const index = SQUtils.ModelUtils.indexOf(model, "name", tagType + root.communityName)
+            if(index === -1)
+                return false
+
+            const token = SQUtils.ModelUtils.get(model, index)
+
+            // Some assertions:
+            if(!token.isPrivilegedToken)
+                return false
+
+            if(token.isOwner !== isOwner)
+                return false
+
+            // Deploy state check:
+            if(token.deployState !== Constants.ContractTransactionStatus.Completed)
+                return false
+
+            // Token deployed!!
+            return true
+        }
+
+        model: root.community.communityTokens
+
+        onRevisionChanged: {
```

Comment:
Alternatively you could limit number of operations on the model to just one call and do the rest on plain js array:
```
const data = ModelUtils.modelToArray(model, ["name", "isPrivilegedToken", "isOwner"]
ownerOrTMasterTokenItemsExist = data.some(token => {
    return token.name === PermissionsHelpers.ownerTokenNameTag + root.communityName ||
           token.name === PermissionsHelpers.tMasterTokenNameTag + root.communityName
})
isOwnerTokenDeployed = data.some(token => (condition for owner here))
isTMasterTokenDeployed = data.some(token => (condition for tmaster here))
```

Reply:
I'll keep it as it is right now bc it can probably change a bit with the backend integration since maybe instead of checking the `name` we will only check a new role `tokenRole/type`. I'll have in mind for the future!